src/moonlink/src/union_read/read_state_manager.rs

```diff
@@ -145,12 +145,12 @@ impl ReadStateManager {
                     current_snapshot_lsn
                 };
 
-            let read_output = table_state_snapshot.request_read().await?;
+            let snapshot_read_output = table_state_snapshot.request_read().await?;
 
             self.last_read_lsn.store(effective_lsn, Ordering::Release);
-            *last_read_snapshot_output_guard = Arc::new(read_output);
+            *last_read_state_guard = snapshot_read_output.take_as_read_state().await;
```

Comment:
I recall why I cache `ReadOutput` instead of `ReadState` in the previous PR: `take_as_read_state` could involve IO operation if cache miss, but not a problem because currently assume never goes out of disk and never do proactive eviction.

Reply:
it might be ok, because likely different `scan_table`'s asking for the same LSN